aten/src/ATen/functorch/BatchRulesModules.cpp

```diff
@@ -246,7 +246,7 @@ static Tensor one_hot_decomposition_hack(const Tensor &self, int64_t num_classes
     //   TORCH_CHECK(num_classes > self.max().item().toLong(), "Class values must be smaller than num_classes.");
     // }
 
-    shape.push_back(num_classes);
+    shape.emplace_back(num_classes);
     Tensor ret = at::zeros_symint(shape, self.options());
```

Comment:
```suggestion
    Tensor ret = at::zeros_symint(std::move(shape), self.options());
```
If it's an std::vector

Reply:
It's better to review these changes in the third PR.

Reply:
The move is useless because the moved to parameter is array_ref.